wled00/e131.cpp

```diff
@@ -142,32 +142,34 @@ void handleE131Packet(e131_packet_t* p, IPAddress clientIP, byte protocol){
       return;  // nothing to do
       break;
 
-    case DMX_MODE_SINGLE_RGB:   // 3 channel: [R,G,B]
+    case DMX_MODE_SINGLE_RGB:   // 3-4 channels: [R,G,B] + W (if present)
       if (uni != e131Universe) return;
-      if (availDMXLen < 3) return;
+      uint16_t dmxChannelCount;
+      dmxChannelCount = strip.hasWhiteChannel() ? 4 : 3;
```

Comment:
This may not be correct approach as user may choose to allow realtime data only on main segment. And main segment may not contain white channel.
Please check `setRealtimePixel()` for behaviour.

You will need to differentiate according to `useMainSegmentOnly` global variable.

Reply:
Ah alright, I'll further test this the following days according to your comments.

Is `strip.hasWhiteChannel()` always `== true` for RGBW (and ` == false` for RGB) strip only segments?

Due to the utilization of segments to handle different strip types (RGB|RGBW) I think your right and it must be implemented differently. The `strip.hasWhiteChannel()` must be wrong. I haven't had this in mind .. 

Now that I've looked into `setRealtimePixel` method, I'm looking forward to directly implement realtime _Segment (D)RGB(W)_  modes, utilizing `Segment::hasWhite()` and `Segment::setPixelColor(pix, r, g, b, w)` method. 

Due to your sentences

> .. as WLED allows realtime data to be displayed on main segment only, .. 
> .. as user may choose to allow realtime data only on main segment ..

I'm confused. Maybe lack of knowledge of the main segment concept. Is it possible to call for example `Segment &seg = strip.getSegment(1)` and afterwards call `seg.setPixelColor(pix, r, g, b, w)` to realtime set pixels of segment 1? Actually I'm also unclear about this realtime lock statement
```
realtimeLock(realtimeTimeoutMs, mde);
if (realtimeOverride && !(realtimeMode && useMainSegmentOnly)) return;
```
For realtime _Segment (D)RGB(W)_ just leave out the `&& useMainSegmentOnly` is okay?

Maybe I'm wrong, but unfortunately can't find out much about the main segment usage in general.

Nevertheless I'll rebase to 0_15 branch, or maybe provide another PR. If so, I'll close this one.

Reply:
Briefly as I do not have much time today:
- only one segment can be *main* segment, there is always one *main* segment
- any segment can be *main* segment (differentiated with a lighter background in UI)
- normally you do not need to care which segment is *main*
- each segment can contain/encompass outputs of different LED types
- segment keeps track of output capabilities (RGB, W, CCT, etc)
- `strip`'s capabilities are a union of all segment's capabilities
- realtime code can either use whole strip (disregarding segments) or *main* segment
- when using strip (`useMainSegmentOnly!=true`) `strip.hasWhiteChannel()` can be used to detect presence of white channel
- when using main segment (`useMainSegmentOnly==true`) use `mainseg.hasWhite()` (where `Segment &mainseg = strip.getMainSegment();`)
- realtime locking happens automatically when realtime data is received or if instructed so by external command
- realtime lock is released automatically when no data is received in certain amount of time unless externally locked
- relatime lock insures pixels are not updated by internal effects

Reply:
latest commit shall differentiate according to useMainSegmentOnly global variable.